### DISCUSSION TITLE
how to search for at_check using rg?

### DISCUSSION THREAD

**Discussion by monajalal** [BurntSushi/ripgrep#1755] [General] [answered]
I want to know how can I use rg to search all files and subfolders' files within a directory where we have an underline in the item I am searching?

**Comment by monajalal** [comment, marked as answer]
(base) mona@mona:~/research/phosa/external/multiperson/neural_renderer$ rg at_check
(base) mona@mona:~/research/phosa/external/multiperson/neural_renderer$ rg AT_CHECK
neural_renderer/cuda/load_textures_cuda.cpp
15:#define CHECK_CUDA(x) AT_CHECK(x.type().is_cuda(), #x " must be a CUDA tensor")
16:#define CHECK_CONTIGUOUS(x) AT_CHECK(x.is_contiguous(), #x " must be contiguous")

neural_renderer/cuda/create_texture_image_cuda.cpp
13:#define CHECK_CUDA(x) AT_CHECK(x.type().is_cuda(), #x " must be a CUDA tensor")
14:#define CHECK_CONTIGUOUS(x) AT_CHECK(x.is_contiguous(), #x " must be contiguous")

neural_renderer/cuda/rasterize_cuda.cpp
69:#define CHECK_CUDA(x) AT_CHECK(x.type().is_cuda(), #x " must be a CUDA tensor")
70:#define CHECK_CONTIGUOUS(x) AT_CHECK(x.is_contiguous(), #x " must be contiguous")


I realized it was case sensitive. I am all set.

**Reply by BurntSushi** [reply]
Consider using the `-S/--smart-case` flag, which will make searches using patterns with all lowercase automatically case insensitive. Alternatively, the `-i/--case-insensitive` flag makes the search case insensitive unconditionally.

Also, I note that your answer seems to be for a different question then what was asked. You asked how to search for "an underline in the file name," but your answer shows how to search for the string `AT_CHECK` in the contents of a file.
